packages/plugins/rubygem-smart_proxy_ansible/rubygem-smart_proxy_ansible.spec

```diff
@@ -1,66 +1,95 @@
+# template: smart_proxy_plugin
+%{?scl:%scl_package rubygem-%{gem_name}}
+%{!?scl:%global pkg_name %{name}}
+
 %global gem_name smart_proxy_ansible
+%global plugin_name ansible
+
+%{!?_root_datadir:%global _root_datadir %{_datadir}}
+%{!?_root_localstatedir:%global _root_localstatedir %{_localstatedir}}
+%{!?_root_sysconfdir:%global _root_sysconfdir %{_sysconfdir}}
 
-%global foreman_proxy_dir /usr/share/foreman-proxy
-%global foreman_proxy_statedir %{_localstatedir}/lib/foreman-proxy
+%global foreman_proxy_min_version 1.24
+%global foreman_proxy_dir %{_root_datadir}/foreman-proxy
+%global foreman_proxy_statedir %{_root_localstatedir}/foreman-proxy
 %global foreman_proxy_bundlerd_dir %{foreman_proxy_dir}/bundler.d
-%global foreman_proxy_settingsd_dir %{_sysconfdir}/foreman-proxy/settings.d
-%global smart_proxy_dynflow_bundlerd_dir %{?rhel:/opt/theforeman/tfm/root/}%{_datadir}/smart_proxy_dynflow_core/bundler.d
+%global foreman_proxy_settingsd_dir %{_root_sysconfdir}/foreman-proxy/settings.d
+%global smart_proxy_dynflow_bundlerd_dir %{!?scl:/opt/theforeman/tfm/root}%{_datadir}/smart_proxy_dynflow_core/bundler.d
```

Comment:
This is still incorrect. Regardless of whether this package is built in SCL, the `smart_proxy_dynflow_core/bundler.d` directory is in SCL if the platform supports SCL. This probably means it's incorrect for EL8 though where `%{rhel}` will be true but we won't use SCL.

Reply:
As my testing, and understanding goes, this should say "if SCL is not defined, prepend /opt/theforeman/tfm/root; if SCL is defined, don't prepend". If building for SCL, `%{_datadir}` should include the `/opt/theforeman/tfm/root` path. I think the below is correct:


```
ehelms@war tmp (update-smart-proxy-plugins)$ wget http://koji.katello.org/kojifiles/work/tasks/2048/232048/rubygem-smart_proxy_ansible-3.0.1-2.fm1_24.el7.noarch.rpm
--2019-09-27 07:40:35--  http://koji.katello.org/kojifiles/work/tasks/2048/232048/rubygem-smart_proxy_ansible-3.0.1-2.fm1_24.el7.noarch.rpm
Resolving koji.katello.org (koji.katello.org)... 34.224.159.44
Connecting to koji.katello.org (koji.katello.org)|34.224.159.44|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 23488 (23K) [application/x-rpm]
Saving to: ‘rubygem-smart_proxy_ansible-3.0.1-2.fm1_24.el7.noarch.rpm’

rubygem-smart_proxy_ansibl 100%[=====================================>]  22.94K  --.-KB/s    in 0.02s   

2019-09-27 07:40:35 (1.08 MB/s) - ‘rubygem-smart_proxy_ansible-3.0.1-2.fm1_24.el7.noarch.rpm’ saved [23488/23488]

ehelms@war tmp (update-smart-proxy-plugins)$ rpm -qlp rubygem-smart_proxy_ansible-3.0.1-2.fm1_24.el7.noarch.rpm 
/etc/foreman-proxy/ansible.cfg
/etc/foreman-proxy/settings.d/ansible.yml
/opt/theforeman/tfm/root/usr/share/smart_proxy_dynflow_core/bundler.d/foreman_ansible_core.rb
/usr/share/foreman-proxy/.ansible
/usr/share/foreman-proxy/.ansible.cfg
/usr/share/foreman-proxy/.ansible_galaxy
/usr/share/foreman-proxy/bundler.d/smart_proxy_ansible.rb
/usr/share/gems/gems/smart_proxy_ansible-3.0.1
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/LICENSE
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/bin
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/bin/json_inventory.sh
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible.rb
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible/api.rb
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible/exception.rb
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible/http_config.ru
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible/plugin.rb
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible/roles_reader.rb
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible/variables_extractor.rb
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/lib/smart_proxy_ansible/version.rb
/usr/share/gems/gems/smart_proxy_ansible-3.0.1/settings.d
/usr/share/gems/specifications/smart_proxy_ansible-3.0.1.gemspec
/var/foreman-proxy/ansible
/var/foreman-proxy/ansible_galaxy
```

Reply:
I shouldn't review this thing at the start of my day. You're right, at least on EL7. I'm fine with punting EL8 support to later. Maybe EL8 is a good excuse to properly sort out the whole smart_proxy_dynflow_core situation.